cdb2api/cdb2api.c

```diff
@@ -6867,6 +6967,13 @@ int cdb2_open(cdb2_hndl_tp **handle, const char *dbname, const char *type,
         rc = cdb2_get_dbhosts(hndl);
         if (rc)
             debugprint("cdb2_get_dbhosts returns %d\n", rc);
+        if (rc == 0 && hndl->num_shards > 0) {
+            char db_shard[DBNAME_LEN];
+            int shard_num = (hndl->num_shards_sameroom > 0 && (hndl->flags & CDB2_RANDOMROOM)) ? rand() % hndl->num_shards_sameroom : rand() % hndl->num_shards;
```

Comment:
what happens if the client uses CDB2_ROOM instead of CDB2_RANDOMROOM? do I get a node from same room?